bootstrap/service.go

```diff
@@ -152,8 +154,10 @@ func (bs bootstrapService) Add(ctx context.Context, token string, cfg Config) (C
 
 	saved, err := bs.configs.Save(ctx, cfg, toConnect)
 	if err != nil {
+		// If id is empty, then a new thing has been created function - bs.thing(id, token)
```

Comment:
// If the ID is empty, a new thing has been created by `bs.thing(id, token)`, so if the bootstrap config creation returns an error, try to delete the newly created thing.

Reply:
Changed